lib/events/dynamic.go

```diff
@@ -0,0 +1,372 @@
+		var e events.GithubConnectorDelete
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case OIDCConnectorCreatedEvent:
+		var e events.OIDCConnectorCreate
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case OIDCConnectorDeletedEvent:
+		var e events.OIDCConnectorDelete
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case SAMLConnectorCreatedEvent:
+		var e events.SAMLConnectorCreate
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case SAMLConnectorDeletedEvent:
+		var e events.SAMLConnectorDelete
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case SessionRejectedEvent:
+		var e events.SessionReject
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case AppSessionStartEvent:
+		var e events.AppSessionStart
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case AppSessionChunkEvent:
+		var e events.AppSessionChunk
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case AppSessionRequestEvent:
+		var e events.AppSessionRequest
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case DatabaseSessionStartEvent:
+		var e events.DatabaseSessionStart
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case DatabaseSessionEndEvent:
+		var e events.DatabaseSessionEnd
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case DatabaseSessionQueryEvent:
+		var e events.DatabaseSessionQuery
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case KubeRequestEvent:
+		var e events.KubeRequest
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case MFADeviceAddEvent:
+		var e events.MFADeviceAdd
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	case MFADeviceDeleteEvent:
+		var e events.MFADeviceDelete
+		if err := json.Unmarshal(data, &e); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &e, nil
+	default:
+		return nil, trace.BadParameter("unknown event type: %v", eventType)
+	}
+}
+
+// GetSessionID pulls the session ID from the events that have a
+// SessionMetadata. For other events an empty string is returned.
+func GetSessionID(event AuditEvent) (string, error) {
+	fields, err := ToEventFields(event)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	return fields.GetString(SessionEventID), nil
+}
```

Comment:
All audit events that have session IDs implement the [`SessionMetadataGetter`](https://github.com/gravitational/teleport/blob/4284fc3586edff859ee0fd4a82ad7cd499dac413/lib/events/api.go#L436-L439) interface.  Use that instead.  Ex:

```golang
	var sessionID string
	if g, ok := event.(SessionMetadataGetter); ok {
		sessionID = g.GetSessionID()
	}
```